### PR TITLE
feat(protocol): upgrade to use  OZ 4.9.6

### PR DESCRIPTION
**Pull request by dantaik**

### CODE REVIEW

File: packages/protocol/test/L1/gov/TaikoGovernor.t.sol [hunk 1]
```diff
@@ -99,11 +99,6 @@ contract TestTaikoGovernor is TaikoL1TestBase {
             true,
             "Incorrect supports interface"
         );
-        assertEq(
-            taikoGovernor.supportsInterface(type(IGovernorUpgradeable).interfaceId),
```
Comment (dantaik) [inline packages/protocol/test/L1/gov/TaikoGovernor.t.sol:103]: no longer supported this interface in the OZ 4.9.6

**Comment by adaki2004** [comment]
@dantaik We have gone through these (tho 4.9.5) and i guess code size (build fails) can be done and be inspired from these changes:
https://github.com/taikoxyz/taiko-mono/pull/15864/files

File: packages/protocol/contracts/L1/gov/TaikoGovernor.sol [hunk 2]
```diff
@@ -124,6 +124,23 @@ contract TaikoGovernor is
         return 1_000_000_000 ether / 10_000; // 0.01% of Taiko Token
     }
 
+    /// @dev Cancel a proposal with GovernorBravo logic.
```
Comment (dantaik) [inline packages/protocol/contracts/L1/gov/TaikoGovernor.sol:127]: We have an issue with contract size.

> `TaikoGovernor` is above the contract size limit (25229 > 24576).

Not sure how we can get round of this issue.

Comment (dantaik) [reply packages/protocol/contracts/L1/gov/TaikoGovernor.sol:127]: Asked OZ.

Comment (adaki2004) [reply packages/protocol/contracts/L1/gov/TaikoGovernor.sol:127]: Also applied (locally) what i suggested, and still above the threshold..

Comment (adaki2004) [reply packages/protocol/contracts/L1/gov/TaikoGovernor.sol:127]: Ok, i have a workaround for the size issue : https://github.com/taikoxyz/taiko-mono/pull/16363

**Review by Brechtpd — APPROVED** [review]

File: packages/protocol/contracts/L1/gov/TaikoGovernor.sol [hunk 3]
```diff
@@ -1,22 +1,20 @@
...
-import "@openzeppelin/contracts-upgradeable/governance/GovernorUpgradeable.sol";
 import
     "@openzeppelin/contracts-upgradeable/governance/compatibility/GovernorCompatibilityBravoUpgradeable.sol";
-import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorVotesUpgradeable.sol";
 import
     "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorVotesQuorumFractionUpgradeable.sol";
 import
     "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorTimelockControlUpgradeable.sol";
-import "../../common/EssentialContract.sol";
+
```
Comment (d1onys1us) [inline packages/protocol/contracts/L1/gov/TaikoGovernor.sol:10]:
```suggestion
```

**Review by dionysuzx — APPROVED** [review]

**Pull request merged**